src/remoted/secure.c

```diff
@@ -471,3 +507,87 @@ int _close_sock(keystore * keys, int sock) {
 
     return retval;
 }
+
+int key_request_connect() {
+#ifndef WIN32
+    return OS_ConnectUnixDomain(isChroot() ? WM_KEY_REQUEST_SOCK : WM_KEY_REQUEST_SOCK_PATH, SOCK_DGRAM, OS_MAXSTR);
+#else
+    return -1;
+#endif
+}
+
+static int send_key_request(int socket,const char *msg) {
+    return OS_SendUnix(socket,msg,strlen(msg));
+}
+
+static void _push_request(const char *request,const char *type) {
+    char *msg = NULL;
+    static pthread_mutex_t key_request_mutex = PTHREAD_MUTEX_INITIALIZER;
+
+    os_calloc(OS_MAXSTR,sizeof(char),msg);
+    snprintf(msg,OS_MAXSTR,"%s:%s",type,request);
+
+    w_mutex_lock(&key_request_mutex);
+    if (!OSHash_Get_ex(key_request_hash, msg)) {
```

Comment:
The agent-key-polling module already avoids duplicate requests. We should not do that twice.